storage/simulation/cluster.go

```diff
@@ -169,15 +177,20 @@ func (c *Cluster) splitRange(rangeID roachpb.RangeID) {
 // runEpoch steps through a single instance of the simulator. Returns true if no
 // actions were performed during this epoch.
 // Each epoch performs the following steps:
-// 1) The status of every store is gossiped so the store pool is up to date.
-// 2) Each replica on every range calls the allocator to determine if there are
+// 1) Perform all scripted actions.
+// 2) The status of every store is gossiped so the store pool is up to date.
+// 3) Each replica on every range calls the allocator to determine if there are
 //    any actions required.
-// 3) The replica on each range with the highest priority executes it's action.
-// 4) The rangesByStore map is recalculated.
-// 5) The current status of the cluster is output.
+// 4) The replica on each range with the highest priority executes it's action.
+// 5) The rangesByStore map is recalculated.
+// 6) The current status of the cluster is output.
+// 7) Flush the outputs.
 func (c *Cluster) runEpoch() bool {
 	c.epoch++
 
+	// Perform all the requested scripted actions.
+	completed := c.executeScriptedActions()
```

Comment:
this doesn't look right. if the script results in the termination of the sim, why are you still doing all the other below (including running an extra action cycle)?